proto/babylon/btcstaking/v1/events.proto

```diff
@@ -40,11 +40,19 @@ message EventPowerDistUpdate {
     bytes pk = 1 [ (gogoproto.customtype) = "github.com/babylonlabs-io/babylon/types.BIP340PubKey" ];
   }
 
+  // EventJailedFinalityProvider defines an event that a finality provider
+  // is jailed after being detected sluggish
+  message EventJailedFinalityProvider {
+    bytes pk = 1 [ (gogoproto.customtype) = "github.com/babylonlabs-io/babylon/types.BIP340PubKey" ];
+  }
+
   // ev is the event that affects voting power distribution
   oneof ev {
     // slashed_fp means a finality provider is slashed
     EventSlashedFinalityProvider slashed_fp = 1;
+    // jailed_fp means a finality provider is jailed
+    EventJailedFinalityProvider jailed_fp = 2;
```

Comment:
We will have an unjailed event in subsequent PRs right?

Reply:
I don't think so because for unjailing, we just need to revert the `jailed` flag in the finality provider and voting power will be assigned. In [a1b253d](https://github.com/babylonlabs-io/babylon/pull/65/commits/a1b253d42261366bb623674f0b611d4cf49df885), we used the same approach as we did in the timestamping pub rand where `isJailed` is introduced to the fp dist info and if `isJailed` is true, the voting power will be hidden. In another word, `jailing` should not affect delegation accumulation.

Reply:
Under which condition will `IsJailed` be reverted? This is slightly different from `IsTimestamped` which won't be reverted once set `true`

Reply:
we update `IsJailed` of each fp dist info of the cache within `recordVotingPowerAndCache` like we did for updating the timestamping status. See https://github.com/babylonlabs-io/babylon/blob/a1b253d42261366bb623674f0b611d4cf49df885/x/btcstaking/keeper/power_dist_change.go#L100